internal/cmd/skupper/debug/nonkube/debug.go

```diff
@@ -21,13 +45,355 @@ func NewCmdDebug() *CmdDebug {
+			err := utils.WriteObject(listener, path+"Listener-"+listener.Name, tw)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	connectors, err := cmd.connectorHandler.List(opts)
+	if err == nil && connectors != nil && len(connectors) != 0 {
+		for _, connector := range connectors {
+			err := utils.WriteObject(connector, path+"Connector-"+connector.Name, tw)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	secrets, err := cmd.secretHandler.List(opts)
+	if err == nil && secrets != nil && len(secrets) != 0 {
+		for _, secret := range secrets {
+			err := utils.WriteObject(secret, path+"Secret-"+secret.Name, tw)
+			if err != nil {
+				return err
+			}
+		}
+	}
 
-func (cmd *CmdDebug) InputToOptions() {}
+	links, err := cmd.linkHandler.List(opts)
+	if err == nil && links != nil && len(links) != 0 {
+		for _, link := range links {
+			err := utils.WriteObject(link, path+"Link-"+link.Name, tw)
+			if err != nil {
+				return err
+			}
+		}
+	}
 
-func (cmd *CmdDebug) Run() error { fmt.Println("not yet implemented"); return nil }
+	//internal/scripts
+	path = intpath + "scripts/"
+	scriptsPath := api.GetDefaultOutputPath(cmd.namespace) + "/" + string(api.ScriptsPath)
+	scripts, err := os.ReadDir(scriptsPath)
+	if err == nil && scripts != nil && len(scripts) != 0 {
+		for _, script := range scripts {
+			fileName := script.Name()
+			file, err := os.ReadFile(scriptsPath + "/" + fileName)
+			if file != nil && err == nil {
+				utils.WriteTar(path+fileName, file, time.Now(), tw)
+			}
+		}
+	}
+
+	// runtime/certs
+	path = rpath + "certs/"
+	certPath = api.GetDefaultOutputPath(cmd.namespace) + "/" + string(api.CertificatesPath)
+	certDirs, err := os.ReadDir(certPath)
+	if err == nil && certDirs != nil && len(certDirs) != 0 {
+		for _, certDir := range certDirs {
+			for x := range certFiles {
+				fileName := certDir.Name() + "/" + certFiles[x]
+				file, err := os.ReadFile(certPath + "/" + fileName)
+				if file != nil && err == nil {
+					utils.WriteTar(path+fileName, file, time.Now(), tw)
+				}
+			}
+		}
+	}
+
+	// runtime/issuers
+	path = rpath + "issuers/"
+	iPath := api.GetDefaultOutputPath(cmd.namespace) + "/" + string(api.IssuersPath)
+	iDirs, err := os.ReadDir(iPath)
+	if err == nil && iDirs != nil && len(iDirs) != 0 {
+		for _, iDir := range iDirs {
+			for x := range certFiles {
+				fileName := iDir.Name() + "/" + certFiles[x]
+				file, err := os.ReadFile(iPath + "/" + fileName)
+				if file != nil && err == nil {
+					utils.WriteTar(path+fileName, file, time.Now(), tw)
+				}
+			}
+		}
+	}
+
+	// runtime/links
+	path = rpath + "links/"
+	lpath := api.GetDefaultOutputPath(cmd.namespace) + "/" + string(api.RuntimeTokenPath)
+	lopts := fs.GetOptions{ResourcesPath: lpath, RuntimeFirst: true, LogWarning: false}
+	links, err = cmd.linkHandler.List(lopts)
+	if err == nil && links != nil && len(links) != 0 {
+		for _, link := range links {
+			err := utils.WriteObject(link, path+link.Name+link.Spec.Endpoints[0].Host, tw)
```

Comment:
A dash is missing between `link.Name` and `link.Spec.Endpoints[0].Host`.

Reply:
This one is still missing.

Reply:
fixed